exporter/elasticsearchexporter/pdata_serializer.go

```diff
@@ -0,0 +1,381 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package elasticsearchexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter"
```

Comment:
nit: Seems that this serializer is a good candidate to be moved to a separate package, and only expose the `serialize*` funcs.

Reply:
I've tried that in the beginning but there are a few package-private things it needs to access. For example, `dataPoint` and `dataStream*` constants in `attribute.go`, which we'd need to make public.

Reply:
I've moved bufferpool into a separate package now